### PR TITLE
Working on Issue #57

### PR DESCRIPTION
**Pull request by Eddie-CooRo**
I've refactored some of callbacks and promises
For others i strongly recommend you to use this package: [Request-Promise-Native](https://github.com/request/request-promise-native) instead of request lib
take a look and let me know what you think about it

### CODE REVIEW

**Comment by bluzi** [comment]
Overall looks great, real quality work. Thanks!
I agree about `request-promise-native`, in general I prefer using promises over callbacks, especially when using async/await.

I'd be happy to merge it when it's finished, just note the linting errors (you can use `npm run lint` to test it locally, and you can try using `npm run lint:fix` to try and fix it automatically. 

@Eddie-CooRo

**Comment by Eddie-CooRo** [comment]
Good to hear that
I'm working on it
@bluzi

**Comment by bluzi** [comment]
Great

Also, please ignore codeclimate's check, it's a cool service but unfortunately the codebase is not ready for it yet :)

**Comment by Eddie-CooRo** [comment]
I think i'm done!

File: utils/pullrequest.js [hunk 1]
```diff
@@ -35,56 +35,60 @@ const pr = {
   TravisBuddy :green_heart:`,
 };
```
Comment (Eddie-CooRo) [inline utils/pullrequest.js]: You should take a look at this file
I've moved everything inside an async function in order to refactor those nested callbacks

File: utils/utils.js [hunk 2]
```diff
@@ -8,29 +8,35 @@ const GitHub = require('github-api');
...
+    } else {
+      logger.log('Max attempts achived, giving up done');
     }
 
-    logger.log(`Done not found, requesting new log... (${attempts}/${MAX_ATTEMPTS_TO_GET_DONE}`, data);
+    const cleanLog = stripAnsi(log);
+    return cleanLog;
+  }
+
+  logger.log(`Done not found, requesting new log... (${attempts}/${MAX_ATTEMPTS_TO_GET_DONE}`, data);
```
[the remark this answers is not in the document]
Comment (bluzi) [reply utils/utils.js]: I can't actually think of a better salutation myself, but I'll search some more later, see if anything more elegant comes up :)

File: utils/pullrequest.js (hunk 1 above)
Comment (bluzi) [reply utils/pullrequest.js]: Yup, looks good

**Comment by bluzi** [comment]
Thanks again! everything looks great! 
looks like you've missed one here: https://github.com/Eddie-CooRo/travis-buddy/blob/master/resolvers/simple.resolver.js

**Comment by Eddie-CooRo** [comment]
Oops...
i didn't see that

**Comment by bluzi** [comment]
Thanks @Eddie-CooRo, that was super helpful. You've done great job and the code looks great, that's awesome.

If you have time and feel like it, feel free to look at some other issues we have open. If you have questions or need help understanding something please ask. 

Also if you have any insights about the maintainability of the project (tests, linting, etc), that's something I'd like to improve, so please let me know.

Again, thanks!

**Comment by Eddie-CooRo** [comment]
Your welcome
I'll take a look at other issues
and about maintainability.... unfortunately i'm not into these things so much... I wish I could help

**Pull request merged**